urldee901fae6_DocX_personal_timetabling_tgz-pJ8-net_personaltt_main_BenchmarkJ8/src/net/personaltt/main/Benchmark.java

```diff
@@ -0,0 +1,789 @@
+    }
+    
+    public static void test1(int number) {
+        System.out.printf("PROBLEM 1 - %s events\n", number);
+        ProblemDefinition p = bench1(number);
+        printInitialCosts(p);
+        testProblem(p);
+    }
+    
+    public static void test2(int number) {
+        System.out.printf("PROBLEM 2 - %s events\n", number);
+        ProblemDefinition p = bench2(number);
+        printInitialCosts(p);
+        testProblem(p);
+    }
+    
+    public static void test3(int number) {
+        System.out.printf("PROBLEM 3 - %s events\n", number);
+        ProblemDefinition p = bench3(number);
+        printInitialCosts(p);
+        testProblem(p);
+    }
+    
+    public static void test4(int number) {
+        System.out.printf("PROBLEM 4 - %s events\n", number);
+        ProblemDefinition p = bench4(number);
+        printInitialCosts(p);
+        testProblem(p);
+    }
+    
+    public static void testS(int number) {
+        System.out.printf("PROBLEM S - %s number\n", number);
+        ProblemDefinition p = bench5(number);
+        printInitialCosts(p);
+        testProblem(p);
+    }
+    
+    public static void testP(int number) {
+        System.out.printf("PROBLEM P - %s number\n", number);
+        ProblemDefinition p = benchPriority(1,number);
+        printInitialCosts(p);
+        testProblem(p);
+    }    
+
+    private static void printInitialCosts(ProblemDefinition p) {
+           
+        MainSolverState currentSolution = new MainSolverState();
+        currentSolution.init(p.initialSchedule);
+        
+        System.out.println(String.format("INITIAL COST: c:%s p:%s", currentSolution.constraintsCost(), currentSolution.preferenceCost()));
+    }
+    
+ 
+    
+    
+    public static void benchmarkPriority() {
+        List<String> results = new ArrayList<String>();
+        for (int i = 0; i <= 0; i+=1) {
+            int nonPrefCount = 0;
+            for (int j = 0; j < 10; j++) {
+                ProblemDefinition problem = benchPriority(i,50);
+                SolverState state = benchSolve(problem);
+                if (state.getBestSolution().getSchedule().getAllocationOf(1).getStart() != 0) {
+                    nonPrefCount++;
+                }
+            }
+            results.add(String.format("Priority: %s, not preffered count: %s/10", i, nonPrefCount));
+        }
+       
+        for (String string : results) {
+            System.out.println(string);
+        }
+    }
+    
+    /**
+     * Experimental benchmark for determination relation between
+     * number of events and interation in whish is found best solution
+     */
+    public static void benchmarkTerminationCondition() {
+        List<String> results = new ArrayList<String>();
+        for (int j = 100; j <= 1000; j+=100) {
+            ProblemDefinition problem = bench1(j);
+            SolverState state = benchSolve(problem);
+            results.add(String.format("Events: %s, best iteration: %s, end iteration: %s",
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: &#39;getItearation&#39; expression, &#39;getLastBestIteration&#39; expression, j (declared at line 201). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 204. Consider formatting this data with &#39;%d&#39; instead.